nautobot/dcim/forms.py

```diff
@@ -194,8 +194,13 @@ def clean(self):
         # TODO: after Location model replaced Site, which was not a hierarchical model, should we allow users to add a VLAN
         # belongs to the parent Location or the child location of the parent device to the `tagged_vlan` field of the interface?
         elif mode == InterfaceModeChoices.MODE_TAGGED:
-            valid_locations = [None, self.cleaned_data[parent_field].location]
-            invalid_vlans = [str(v) for v in tagged_vlans if v.location not in valid_locations]
+            valid_location = self.cleaned_data[parent_field].location
+            invalid_vlans = [
+                str(v)
+                for v in tagged_vlans
+                if v.locations.without_tree_fields().exists()
+                and not VLANLocationAssignment.objects.filter(location=valid_location, vlan=v).exists()
```

Comment:
Most efficient database queries used here.

`if vlan has a location assigned and there's an association between the vlan and the valid location`

Reply:
Quick question, how are we checking if VLAN is global? Is it that if there are no locations assigned to the vlan, it is automatically global?